src/jit/IntMath32Inl.h

```diff
@@ -1304,3 +1304,122 @@ static void emitConvert(sljit_compiler* compiler, Instruction* instr)
     }
     }
 }
+
+static void emitMemory(sljit_compiler* compiler, Instruction* instr)
+{
+    Operand* operand = instr->operands();
+
+    CompileContext* context = CompileContext::get(compiler);
+    JITMemory* memory = context->compiler->memories()[0];
+
+    sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_MEM1(kContextReg),
+                   OffsetOfContextField(memory0Ptr), SLJIT_IMM,
+                   reinterpret_cast<sljit_sw>(memory->memPtr));
+    sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_MEM1(kContextReg),
+                   OffsetOfContextField(memory0Current), SLJIT_IMM,
+                   memory->limit);
+    sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_MEM1(kContextReg),
+                   OffsetOfContextField(memory0Limit), SLJIT_IMM,
+                   memory->current);
+
+    sljit_sw memType;
+
+    JITArg args[2];
+
+    for (Index i = 0; i < instr->paramCount() + instr->resultCount(); ++i) {
+        operandToArg(operand, args[i]);
+        operand++;
+    }
+
+    switch (instr->opcode()) {
+    case Opcode::I32Load8S:
+    case Opcode::I32Store8:
+        memType = SLJIT_MOV32_S8;
+        break;
+    case Opcode::I32Load8U:
+        memType = SLJIT_MOV32_U8;
+        break;
+    case Opcode::I32Load16S:
+    case Opcode::I32Store16:
+        memType = SLJIT_MOV32_S16;
+        break;
+    case Opcode::I32Load16U:
+        memType = SLJIT_MOV32_U16;
+        break;
+    case Opcode::I32Load:
+    case Opcode::I32Store:
+        memType = SLJIT_MOV32;
+        break;
+    case Opcode::I64Load8S:
+    case Opcode::I64Store8:
+        memType = SLJIT_MOV_S8;
+        break;
+    case Opcode::I64Load8U:
+        memType = SLJIT_MOV_U8;
+        break;
+    case Opcode::I64Load16S:
+    case Opcode::I64Store16:
+        memType = SLJIT_MOV_S16;
+        break;
+    case Opcode::I64Load16U:
+        memType = SLJIT_MOV_U16;
+        break;
+    case Opcode::I64Load32S:
+    case Opcode::I64Store32:
+        memType = SLJIT_MOV_S32;
+        break;
+    case Opcode::I64Load32U:
+        memType = SLJIT_MOV_U32;
+        break;
+    case Opcode::I64Load:
+    case Opcode::I64Store:
+        memType = SLJIT_MOV;
+        break;
+    case Opcode::MemorySize:
+        sljit_emit_op1(compiler, SLJIT_MOV, args[0].arg, args[0].argw, SLJIT_IMM, memory->current);
+        return;
+    case Opcode::MemoryGrow:
+#if 0
```

Comment:
if 0?

Reply:
This is only temporary, it was just used for the sake of testing and as the patch is not ready yet I left it in.